### PR TITLE
Simple Vertically partitioned SplitNN

### LINKED ISSUE #6: Simple Vertically partitioned SplitNN
## Feature Description
Develop a SplitNN with a single input head, using `syft` to send the model parts to different devices.

See the [PySyft tutorials](https://github.com/OpenMined/PySyft/tree/master/examples/tutorials/advanced/split_neural_network) for a base SplitNN; we may be able to use this SplitNN exactly

Place the model in `src/splitnn.py`

## Additional Context
Part of EPIC #3

### PR DESCRIPTION
**Pull request by pavlos-p**
## Description
Simple Vertically partitioned SplitNN.

Alice worker has two segments of the model and the MNIST Images.

Bob worker has a segment of the model and the MNIST Labels.

Closes #6 

## How has this been tested?
Tested using the Jupyter Notebook of the [SplitNN Tutorial 3](https://github.com/OpenMined/PySyft/blob/master/examples/tutorials/advanced/split_neural_network/Tutorial%203%20-%20Folded%20Split%20Neural%20Network.ipynb)

**After the training:**
`print(images)` is pointing to Alice worker and `alice._objects` confirming this.
`print(labels)` is pointing to Bob worker and `bob._objects` confirming this.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/splitnn.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                nn.Linear(hidden_sizes[0], hidden_sizes[1]),
+                nn.ReLU(),
+    ),
+    nn.Sequential(
+                nn.Linear(hidden_sizes[1], output_size),
+                nn.LogSoftmax(dim=1)
+    )
+]
+
+# Create optimisers for each segment and link to them
```
Comment (TTitcombe) [inline src/splitnn.py:94]: I would remove everything from this line down. During our experiments we'll want to define optimizers and workers so we can try different datasets.

Could you create a jupyter notebook in the `examples` directory which creates a defines optimizers and workers and then creates a `SplitNN`, to demonstrate how to use the model

File: src/splitnn.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+alice = sy.VirtualWorker(hook, id="alice")
+bob = sy.VirtualWorker(hook, id="bob")
+
+# Send Model Segments to model locations
+model_locations = [alice, alice, bob]
+for model, location in zip(models, model_locations):
+    model.send(location)
+
+#Instantiate a SpliNN class with our distributed segments and their respective optimizers
+splitNN = SplitNN(models, optimizers)
```
Comment (TTitcombe) [inline src/splitnn.py:110]: the file needs an empty line at the end

File: src/splitnn.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+# Define our model segments
+
+input_size = 784
+hidden_sizes = [128, 640]
+output_size = 10
+
+models = [
+    nn.Sequential(
+                nn.Linear(input_size, hidden_sizes[0]),
```
Comment (TTitcombe) [inline src/splitnn.py:81]: In the simple case we just need two workers, alice and bob. You can combine the first two model parts into a single `nn.Sequential` object

File: src/splitnn.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            i+=1
+        
+        a.append(models[i](remote_a[-1]))
+        self.a = a
+        self.remote_a = remote_a
+        
+        return a[-1]
+    
+    def backward(self):
+        a=self.a
```
Comment (TTitcombe) [inline src/splitnn.py:49]: Can we use more descriptive variables names. Maybe `grads`?

File: src/splitnn.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import syft as sy
+hook = sy.TorchHook(torch)
+
+class SplitNN:
+    def __init__(self, models, optimizers):
+        self.models = models
+        self.optimizers = optimizers
+        
+    def forward(self, x):
+        a = []
```
Comment (TTitcombe) [inline src/splitnn.py:22]: Same comment as in the `backward` method r.e. variable names
